### PR TITLE
Add Snyk scanning & monitoring

### PR DESCRIPTION
**Pull request by ryan-wren**

### CODE REVIEW

**Review by ssyms — APPROVED** [review]

**Review by vallieres — APPROVED** [review]

File: .circleci/config.yml [hunk 1]
```diff
@@ -83,13 +91,54 @@ jobs:
...
             docker push circleci/circleci-inferconfig:0.$CIRCLE_BUILD_NUM
             docker push circleci/circleci-inferconfig:latest
+  vulnerability-scan:
+    executor: go
+    steps:
+      - checkout
+      - run:
+          name: Setup Scanning
+          command: |
+              git config --global url."https://$GITHUB_USER:$GITHUB_TOKEN@github.com/circleci/".insteadOf "https://github.com/circleci/"
```
Comment (skaasten) [inline .circleci/config.yml:101]: Is this step still needed for a public repo like this (with no private libs being used)?

File: .circleci/config.yml [hunk 2]
```diff
@@ -83,13 +91,54 @@ jobs:
...
 workflows:
   test-and-release:
     jobs:
       - test
       - infer-own-config
       - validate-inferred-configs
+      - vulnerability-scan:
+          context: org-global-employees
       - build-and-push-docker-image:
           requires:
```
Comment (skaasten) [inline .circleci/config.yml:143]: Do we want to still push docker images if the scan job fails?

Comment (ryan-wren) [reply .circleci/config.yml:143]: No. I'll make the `build-and-push-docker-image` job require the `vulnerability-scan` job.

File: .circleci/config.yml (hunk 1 above)
Comment (ryan-wren) [reply .circleci/config.yml:101]: I don't think so. I'll remove this step.

**Review by skaasten — APPROVED** [review]

**Pull request merged**